core/api/src/adapter.rs

```diff
@@ -253,51 +254,72 @@ where
         state_root: Hash,
     ) -> ProtocolResult<EthAccountProof> {
         let state_mpt_tree = MPTTrie::from_root(state_root, Arc::clone(&self.trie_db))?;
-        let raw_account = state_mpt_tree
-            .get(address.as_bytes())?
-            .ok_or_else(|| APIError::Adapter("Can't find this address".to_string()))?;
-
-        let account = Account::decode(raw_account).unwrap();
-
-        let account_proof = state_mpt_tree
+        let account_proof: Vec<Hex> = state_mpt_tree
             .get_proof(address.as_bytes())?
             .into_iter()
             .map(Hex::encode)
             .collect();
-
-        let storage_mpt_tree = MPTTrie::from_root(account.storage_root, Arc::clone(&self.trie_db))?;
-
-        let mut storage_proofs = Vec::with_capacity(storage_position.len());
-
-        for h in storage_position {
-            let hash: Hash = BigEndianHash::from_uint(&h);
-            let proof = EthStorageProof {
-                key:   hash,
-                value: storage_mpt_tree
-                    .get(hash.as_bytes())?
-                    .map(|v| H256::from_slice(&v))
-                    .ok_or_else(|| {
-                        Into::<ProtocolError>::into(APIError::Adapter(
-                            "Can't find this position".to_string(),
-                        ))
-                    })?,
-                proof: storage_mpt_tree
-                    .get_proof(hash.as_bytes())?
-                    .into_iter()
-                    .map(Hex::encode)
-                    .collect(),
-            };
-            storage_proofs.push(proof);
+        match state_mpt_tree.get(address.as_bytes())? {
```

Comment:
The error condition of get, I think can be ignored, its only error is that it cannot parse the db data https://docs.rs/cita_trie/latest/src/cita_trie/trie.rs.html#849